lib/compile.ml

```diff
@@ -32,20 +38,63 @@ let compile_fragment all_infos info =
       rule odoc_path
         ~fdeps:(Inputs.input_file info :: dep_odocs)
         [
-          Format.asprintf "odoc compile --package %s $< %s -o $@" info.package
+          Format.asprintf "odoc compile --parent %s $< %s -o $@" parent_page
             include_str;
         ];
       phony_rule ("compile-" ^ info.package) ~fdeps:[ odoc_path ] [];
     ]
 
-let gen inputs =
-  let packages = Inputs.split_packages inputs in
-  let package_rules_s =
-    List.map (fun (pkg, _) -> "compile-" ^ pkg) (StringMap.bindings packages)
+let find_user_index pkg inputs =
+  let index_name = "page-" ^ pkg in
+  List.find_opt (fun inp -> inp.Inputs.name = index_name) inputs
```

Comment:
I think that's not the right way of searching and there may be collisions. It should be looking at file names.